src/selectors/entities/preferences.js

```diff
@@ -77,3 +78,55 @@ export const getTeammateNameDisplaySetting = createSelector(
         return General.TEAMMATE_NAME_DISPLAY.SHOW_USERNAME;
     }
 );
+
+export const getTheme = createSelector(
+    getMyPreferences,
+    getCurrentTeamId,
+    (myPreferences, currentTeamId) => {
+        // Prefer the user's current team-specific theme over the user's current global theme over the default theme
+        let themePreference;
+        if (currentTeamId) {
+            themePreference = myPreferences[getPreferenceKey(Preferences.CATEGORY_THEME, currentTeamId)];
+        }
+
+        if (!themePreference) {
+            themePreference = myPreferences[getPreferenceKey(Preferences.CATEGORY_THEME, '')];
+        }
+
+        let theme;
+        if (themePreference) {
+            theme = themePreference.value;
+        } else {
+            theme = Preferences.THEMES.default;
+        }
+
+        if (typeof theme === 'string') {
+            try {
+                // A custom theme will be a JSON-serialized object stored in a preference
+                theme = JSON.parse(theme);
+            } catch (e) {
+                // But if it's not a JSON object, it must be the name of a default theme
+                theme = Preferences.THEMES[theme];
```

Comment:
Can you remove this part and the theme constants from the redux repo? I need to think more about how this'll work since I kind of jammed it in here half-finished, and it won't work if/when we allow servers to change the default themes (there's a Shift PR in platform to allow just that). Maybe it can still have the default theme here for the error case, but the ability to not store just the string key isn't working yet